src/assetpreviews/Video.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * @link https://craftcms.com/
+ * @copyright Copyright (c) Pixel & Tonic, Inc.
+ * @license https://craftcms.github.io/license/
+ * @since 3.4.x
+ */
+namespace craft\assetpreviews;
+
+use Craft;
+use craft\base\AssetPreviewHandler;
+
+/**
+ * Provides functionality to preview videos
+ *
+ * @author Pixel & Tonic, Inc. <support@pixelandtonic.com>
+ * @since 3.4.x
+ */
+class Video extends AssetPreviewHandler
+{
+    /**
+     * @inheritdoc
+     */
+    public function getPreviewHtml(): string
+    {
+        /** @var Volume $volume */
+        $volume = $this->asset->getVolume();
+
+        if ($volume->hasUrls) {
+            $url = $this->asset->getUrl();
+        } else {
+            $source = $this->asset->getTransformSource();
+            $url = Craft::$app->getAssetManager()->getPublishedUrl($source, true);
```

Comment:
I don’t think we should be downloading video files just for previewing. Can you just throw a \yii\base\NotSupportedException` here instead? (I’ve updated the controller to be prepared for that possibility – 0319e770e6fa78f5140869a7aac955bcbd664398.)

Reply:
That's a good idea!